packages/utilities/src/memoize.ts

```diff
@@ -120,6 +127,10 @@ export function memoizeFunction<T extends (...args: any[]) => RET_TYPE, RET_TYPE
       cacheSize++;
     }
 
+    if (doNotCacheIfCallbackResultIsNull && !currentNode.value) {
```

Comment:
The `!currentNode.value` check is too loose. It will prevent caching of *any falsey value*: `null`, `undefined`, `false`, `0`, `NaN`, and `''` (empty string). Instead you need to check for *specifically* `!== null` and/or `!== undefined` (depending on what you decide the behavior should be).

You should also add tests for this new parameter, including that it does *not* prevent caching of other falsey values.

Reply:
Modified the check  and I have also added a new test to check that it does not prevent caching of falsey values other that null and undefined.